### DISCUSSION TITLE
Analytics for Crypto Prices and Marketcap Changes

### DISCUSSION THREAD

**Discussion by nazliander** [MaterializeInc/mz-hack-day-2022#10] [Show and tell]
I created a dashboard for observing cryptocurrency prices and Marketcap changes every 20 minutes, using the very nice project template.

Here is the link: https://github.com/nazliander/mz-hack-day-2022

⛵ Data come from Coinranking API: https://developers.coinranking.com/api It requires an API KEY.
🔨 I used an older Kafka producer of mine to tweak into a Redpanda producer. The switch was very easy: 
📐 Understanding of staging views vs. materialized views was the hardest point.
🎨 Temporal filters are awesome, not being able to directly use timestamp fields sucks.
🎵 Materialize allows us to use almost all commonly used SQL commands.

Analysis dbt-models: https://github.com/nazliander/mz-hack-day-2022/tree/main/sample_project/dbt/models/examples/marts/crypto

![image](https://user-images.githubusercontent.com/16581813/154582821-7ebced4d-446a-4f3a-a2c5-c30c7e047255.png)

**Comment by morsapaes** [comment]
This is amazing, @nazliander! Your producer is definitely more elegant than mine, ah. Do you plan to write about your project? It would make a really great blogpost!

And thanks again for the feedback:

> 📐 Understanding of staging views vs. materialized views was the hardest point.

Do you think that [this section](https://materialize.com/docs/overview/api-components/#views) of the docs could use some tweaks? I'd love to get your perspective on how this could be better explained; it really is a common source of confusion to new users!

> 🎨 Temporal filters are awesome, not being able to directly use timestamp fields sucks.

Oh, I feel your pain. There's an outstanding issue to improve the usability of `mz_logical_timestamp()`, and [this proposal](https://github.com/MaterializeInc/materialize/issues/7260#issuecomment-971496054) would make things way more intuitive.

**Reply by nazliander** [reply]
> Do you think that [this section](https://materialize.com/docs/overview/api-components/#views) of the docs could use some tweaks? I'd love to get your perspective on how this could be better explained; it really is a common source of confusion to new users!

That section really explains well the materialized view terminology. Thanks a lot for sharing, I benefitted a lot.

> Do you plan to write about your project? It would make a really great blogpost!

Also, thanks a lot for the nice suggestion @morsapaes. By giving some credits to you and the team, I created a blog post about this project.

Here is the link from my website (and [dev.to](https://dev.to/nazliander/realtime-analysis-of-cryptocurrency-prices-using-dbt-materialize-redpanda-metabase-41n2)): https://nander.cc/realtime-analysis-of-crypto-prices-using-dbt-materialize
